app/models/loan_filtered_question.rb

```diff
@@ -0,0 +1,47 @@
+# Inherits from LoanQuestion. Adds a Loan and provides methods relating to the combination of the two.
+class LoanFilteredQuestion < FilteredQuestion
+  attr_accessor :loan
+
+  def initialize(question, loan:)
+    super(question, loan: loan)
+    @loan = loan
+  end
+
+  # Returns child questions that are applicable to the given loan. Sorts by requiredness, then position.
+  def children
+    @children ||= decorated_children.select(&:visible?).sort_by do |i|
+      [i.required? ? 1 : 2, i.position]
+    end
+  end
+
+  # Resolves if this particular question is considered required for the provided loan, based on
+  # presence of association records in the loan_questions_options relation table, and the
+  # 'override_associations' flag.
+  # - If override is true and join records are present, question is required for those loan types
+  #   and optional for all others
+  # - If override is true and no records are present, all are optional
+  # - If override is false, inherit from parent
+  # - Top level nodes (those with depth = 1 are direct children of the root) effectively have
+  #   override always true
+  # Note, loan type association records are ignored for questions without the 'override_assocations'
+  # flag assigned.
+  def required?
+    if override_associations || depth == 1
+      loan_types.include?(loan.loan_type_option)
+    else
+      parent&.required?
```

Comment:
I wasn't able to run specs on 6941 because of this line

Reply:
This is a new (fancy) [operator](http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/) in Ruby 2.3. Looks like our .ruby-version file didn't get updated in this PR? @linesarefuzzy can you fix?

I'm also curious how the specs passed for you without this...

Reply:
Ah, there was no .ruby-version file in this project, so it used my default, which is 2.3.1. I just pushed one. 

This is also equivalent to `parent.try(:required?)`. Should we just stick with that until everyone is upgraded?

Reply:
No I'm fine with the upgrade. Just need the ruby version file.